assessment.md

```diff
@@ -143,39 +143,96 @@ TODO: add description for each assessment level for this skill.
+* Inadequate = Written software is not efficient in nature and often strays into unnecessary complexities; conditional statements are frequently overly convoluted and hard-coding is consistently present throughout.
+* Adequate = Written software is hardly efficient in nature and occasionally strays into unnecessary complexities; conditional statements can occasionally be overly convoluted and hard-coding is frequently present.
+* Good = Written software is decently efficient in nature and barely strays into unnecessary complexities; conditional statements are often not overly convoluted and hard-coding is only semi-present at best.
+* Excellent = Written software is incredibly efficient in nature and does not stray into unnecessary complexities or perhaps enter complexity without reason; conditional statements are not overly convoluted and hard-coding is hardly, if at all present.
 
 * Mitigating software risk
 
-TODO: add description for each assessment level for this skill.
+Ability to plan for and mitigate risks.
+
+* None = Does not attempt to mitigate or identify potential risks. Does not make an effort to communicate with teammates or handle problems.
+* Inadequate = Makes little attempt to mitigate or identify risks. Low effort in communication and planning.
+* Adequate = Makes an attempt to mitigate and identify risks. Communicates with teammates about some risks but may not be adequately prepared.
+* Good = Makes a good attempt at identifying and mitigating potential risks. Takes some early action to prevent problems. Often communicates with teammates about potential issues. Is often able to handle unexpected problems.  
+* Excellent = Identifies potential risks and actively works to avoid them. Communicates with group members to effectively prevent and manage any risks or problems. Is able to handle unexpected problems when they do appear and asks for help when appropriate.
 
 * Software testing
 
-TODO: add description for each assessment level for this skill.
+Understanding and successfully completing software testing through the correct
+installation of the necessary tools needed to test the software:
+
+* None = Does not have the tools installed to test the Software.
+* Inadequate = Has the correct tools installed but does not test the software with them.
+* Adequate = Has the correct tools installed, but only tests some of the Software.
+* Good = Has the correct tools installed and tests more or all of the Software.
+* Excellent = identifies and understands which tools to use to test the software while
+    also providing documentation and advice that helps other team members develop
+    this skill.
 
 * Software refactoring
 
-TODO: add description for each assessment level for this skill.
+Understanding and using refactoring to improve code readability and performance.
+
```

Comment:
These two lines can be removed for consistency with other sections of the document